demisto_sdk/commands/download/downloader.py

```diff
@@ -160,271 +141,316 @@ def __init__(
-            return 1
+        try:
+            if not (self.verify_flags() and self.verify_output_path()):
+                return 1
 
-        self.build_pack_content()
-        self.build_custom_content() if not self.download_system_item else self.build_system_content()
-        self.update_pack_hierarchy()
-        self.merge_into_pack()
-        self.log_files_downloaded()
-        self.log_files_not_downloaded()
+            if self.init:
+                self.initialize_output_path()
 
-        for entry in self.files_not_downloaded:
-            file, reason = entry
-            if reason != FILE_EXIST_REASON:
-                return 1
+            if self.download_system_items and not self.list_files:
+                downloaded_content_objects = self.fetch_system_content()
+
+            else:
+                all_custom_content_data = self.download_custom_content()
+                all_custom_content_objects = self.parse_custom_content_data(custom_content_data=all_custom_content_data)
+
+                # If we're in list-files mode, print the list of all available custom content and exit
+                if self.list_files:
+                    logger.info(f"list-files (-lf) mode detected. Listing available custom content files "
+                                f"({len(all_custom_content_objects)}):")
+                    table_str = self.create_custom_content_table(custom_content_objects=all_custom_content_objects)
+
+                    logger.info(table_str)
+                    return 0
+
+                uuid_mapping = self.create_uuid_to_name_mapping(custom_content_objects=all_custom_content_objects)
+
+                # Filter custom content so that we'll process only downloaded content
+                downloaded_content_objects = self.filter_custom_content(
+                    custom_content_objects=all_custom_content_objects)
+
+                # Replace UUID IDs with names in filtered content (only content we download)
+                changed_uuids_count = 0
+                for _, file_object in downloaded_content_objects.items():
+                    if self.replace_uuid_ids(custom_content_object=file_object, uuid_mapping=uuid_mapping):
+                        changed_uuids_count += 1
+
+                if changed_uuids_count > 0:
+                    logger.debug(f"Replaced UUID IDs in {changed_uuids_count} custom content items.")
 
-        return 0
+            existing_pack_data = self.build_existing_pack_structure(existing_pack_path=Path(self.output_pack_path))
+
+            result = self.write_files_into_output_path(downloaded_content_objects=downloaded_content_objects,
+                                                       existing_pack_structure=existing_pack_data)
+
+            if result:
+                logger.info(f"Content items downloaded successfully into {self.output_pack_path}.")
+                return 0
+
+            logger.error(f"Download process failed.")
```

Comment:
When do we get here? What else can the logs tell users?

Reply:
Fulle code section is:

```python
            result = self.write_files_into_output_path(downloaded_content_objects=downloaded_content_objects,
                                                       existing_pack_structure=existing_pack_data)

            if result:
                logger.info(f"Content items downloaded successfully into {self.output_pack_path}.")
                return 0

            logger.error(f"Download process failed.")
            return 1
```
`self.write_files_into_output_path` returns `True` if writing **all** downloaded files to disk succeeded, and `False` if even one of them failed. Then we use this response to decide what message to log and what return code to return.

Note that `self.write_files_into_output_path` logs beforehand how many items were downloaded successfully, how many were skipped (if they already existed, and `--force` wasn't used), and how many failed.

```python
        summary_log = f"{successful_downloads_count} files were downloaded successfully."

        if failed_downloads_count:
            summary_log += f"\n{failed_downloads_count} files failed to download."

        if existing_files_skipped_count:
            summary_log += f"\n{existing_files_skipped_count} files that exist in the output pack were skipped. " \
                           f"Use the '-f' / '--force' flag to override."
```